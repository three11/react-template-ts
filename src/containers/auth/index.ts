export { default as Login } from './login';
export { default as PasswordReset } from './password-reset';
export { default as Signup } from './signup';
