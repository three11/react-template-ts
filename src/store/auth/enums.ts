export enum AuthActionType {
	LOGIN_REQUEST = 'LOGIN_REQUEST',
	LOGIN_SUCCESS = 'LOGIN_SUCCESS',
	LOGIN_FAILED = 'LOGIN_FAILED',

	LOGOUT_REQUEST = 'LOGOUT_REQUEST',
	LOGOUT_SUCCESS = 'LOGOUT_SUCCESS',
	LOGOUT_FAILED = 'LOGOUT_FAILED',

	SIGNUP_REQUEST = 'SIGNUP_REQUEST',
	SIGNUP_SUCCESS = 'SIGNUP_SUCCESS',
	SIGNUP_FAILED = 'SIGNUP_FAILED',

	PASSWORD_RESET_REQUEST = 'PASSWORD_RESET_REQUEST',
	PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS',
	PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED',

	SET_LOCALE_REQUEST = 'SET_LOCALE_REQUEST',
	SET_LOCALE_FAILED = 'SET_LOCALE_FAILED',
	SET_LOCALE_SUCCESS = 'SET_LOCALE_SUCCESS',

	RESET_AUTH = 'RESET_AUTH'
}
