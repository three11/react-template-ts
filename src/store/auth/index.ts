export * from './enums';
export * from './interfaces';
export * from './reducer';
export { default as authReducer } from './reducer';
export * from './sagas';
