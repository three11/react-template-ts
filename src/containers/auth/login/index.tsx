import * as React from 'react';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { Link, useNavigate } from 'react-router-dom';

import { Routes } from '@utilities';
import { AuthActionType } from '@store/enums';
import { Wrapper, LoginForm } from '@components';

export const Login: React.FunctionComponent = () => {
	const { t } = useTranslation();
	const dispatch = useDispatch();
	const navigate = useNavigate();

	return (
		<Wrapper className="o-wrapper--fancy">
			<LoginForm
				onSubmit={(payload: any): void => {
					dispatch({
						type: AuthActionType.LOGIN_REQUEST,
						payload: {
							...payload,
							redirect: (): void => navigate(Routes.BASE)
						}
					});
				}}
			>
				<p className="c-form__hint">
					{t("Don't have an account?")} <Link to={Routes.SIGNUP}>{t('Sign up')}</Link>
				</p>

				<p className="c-form__hint">
					{t('Forgot password?')} <Link to={Routes.PASSWORD_RESET}>{t('Reset password')}</Link>
				</p>
			</LoginForm>
		</Wrapper>
	);
};

export default Login;
