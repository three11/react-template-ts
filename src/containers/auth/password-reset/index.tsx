import * as React from 'react';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { Link, useNavigate } from 'react-router-dom';

import { Routes } from '@utilities';
import { AuthActionType } from '@store/enums';
import { Wrapper, PasswordResetForm } from '@components';

export const PasswordReset: React.FunctionComponent = () => {
	const { t } = useTranslation();
	const dispatch = useDispatch();
	const navigate = useNavigate();

	return (
		<Wrapper className="o-wrapper--fancy">
			<PasswordResetForm
				onSubmit={(payload: any): void => {
					dispatch({
						type: AuthActionType.PASSWORD_RESET_REQUEST,
						payload: {
							...payload,
							redirect: (): void => navigate(Routes.LOGIN)
						}
					});
				}}
			>
				<p className="c-form__hint">
					{t('You can also')} <Link to={Routes.SIGNUP}>{t('Sign up')}</Link> {t('or')}{' '}
					<Link to={Routes.LOGIN}>{t('Login')}</Link>
				</p>
			</PasswordResetForm>
		</Wrapper>
	);
};

export default PasswordReset;
