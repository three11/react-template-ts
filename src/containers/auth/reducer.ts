import { TOKEN_KEY, REFRESH_TOKEN_KEY, TOKEN_THRESHOLD_KEY } from '@utilities';

import { AuthActionType } from './enums';
import { AuthState, AuthAction } from './interfaces';

export const initialState: AuthState = {
	token: localStorage.getItem(TOKEN_KEY) || '',
	locale: 'en',
	loading: false,
	threshold: Number(localStorage.getItem(TOKEN_THRESHOLD_KEY)) || 0,
	loginError: '',
	logoutError: '',
	signupError: '',
	refreshToken: localStorage.getItem(REFRESH_TOKEN_KEY) || '',
	passwordResetError: ''
};

export default (state = initialState, { type, payload }: AuthAction): AuthState => {
	switch (type) {
		case AuthActionType.LOGIN_REQUEST:
		case AuthActionType.LOGOUT_REQUEST:
		case AuthActionType.SIGNUP_REQUEST:
		case AuthActionType.SET_LOCALE_REQUEST:
		case AuthActionType.PASSWORD_RESET_REQUEST:
			return {
				...state,
				loading: true
			};
		case AuthActionType.LOGIN_FAILED:
		case AuthActionType.LOGOUT_FAILED:
		case AuthActionType.SIGNUP_FAILED:
		case AuthActionType.SET_LOCALE_FAILED:
		case AuthActionType.PASSWORD_RESET_FAILED:
			return {
				...state,
				...payload,
				loading: false
			};
		case AuthActionType.LOGIN_SUCCESS:
		case AuthActionType.LOGOUT_SUCCESS:
		case AuthActionType.SIGNUP_SUCCESS:
		case AuthActionType.SET_LOCALE_SUCCESS:
		case AuthActionType.PASSWORD_RESET_SUCCESS:
			return {
				...state,
				...payload,
				loading: false
			};
		case AuthActionType.RESET_AUTH:
			return initialState;
		default:
			return state;
	}
};
