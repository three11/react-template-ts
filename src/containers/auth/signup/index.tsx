import * as React from 'react';
import { useDispatch } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { Link, useNavigate } from 'react-router-dom';

import { Routes } from '@utilities';
import { AuthActionType } from '@store/enums';
import { Wrapper, SignupForm } from '@components';

export const Signup: React.FunctionComponent = () => {
	const { t } = useTranslation();
	const dispatch = useDispatch();
	const navigate = useNavigate();

	return (
		<Wrapper className="o-wrapper--fancy">
			<SignupForm
				onSubmit={(payload: any): void => {
					dispatch({
						type: AuthActionType.SIGNUP_REQUEST,
						payload: {
							...payload,
							redirect: (): void => navigate(Routes.LOGIN)
						}
					});
				}}
			>
				<p className="c-form__hint">
					{t('Already have an account?')} <Link to={Routes.LOGIN}>{t('Login')}</Link>
				</p>
			</SignupForm>
		</Wrapper>
	);
};

export default Signup;
