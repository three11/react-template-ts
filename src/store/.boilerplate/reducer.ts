import { ExampleActionType } from './enums';
import { ExampleState, ExampleAction } from './interfaces';

export const initialState: ExampleState = {
	error: false,
	loading: true
};

export default (state = initialState, { type, payload }: ExampleAction): ExampleState => {
	switch (type) {
		case ExampleActionType.REQUEST:
			return {
				...state,
				error: false,
				loading: true
			};
		case ExampleActionType.FAILED:
			return {
				...state,
				error: true,
				loading: false
			};
		case ExampleActionType.SUCCESS:
			return {
				...state,
				error: false,
				loading: false
			};

		default:
			return state;
	}
};
