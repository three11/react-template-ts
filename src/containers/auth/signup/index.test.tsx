import * as React from 'react';
import { render } from '@testing-library/react';

import { Signup } from '.';
import { initialState } from '@store/branches/auth/reducer';
import { TestStoreProvider } from '@components';

test('Signup component should render successfully', () => {
	const { asFragment } = render(
		<TestStoreProvider state={{ auth: { ...initialState } }}>
			<Signup />
		</TestStoreProvider>
	);

	expect(asFragment()).toMatchSnapshot();
});
