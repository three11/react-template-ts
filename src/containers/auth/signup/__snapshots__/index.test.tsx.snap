// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component should render successfully 1`] = `
<DocumentFragment>
  <div
    class="o-wrapper o-wrapper--fancy"
  >
    <header
      class="c-header"
    >
      <div
        class="o-shell o-shell--flex"
      >
        <a
          class="c-logo"
          href="/"
        />
        <nav
          class="c-nav"
        >
          <ul>
            <li>
              <a
                class=""
                href="/about"
              >
                About
              </a>
            </li>
            <li>
              <a
                class=""
                href="/login"
              >
                Login
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <main
      class="o-main"
    >
      <form
        class="c-form c-form--signup"
      >
        <h2>
          Signup
        </h2>
        <div
          class="c-form__cols"
        >
          <div
            class="c-form__col"
          >
            <div
              class="c-form__field c-form__field--email"
            >
              <header
                class="c-form__field-head"
              >
                <label
                  for="email"
                >
                  Email Address
                </label>
              </header>
              <input
                id="email"
                name="email"
                placeholder="someone@example.com"
                type="email"
              />
            </div>
            <div
              class="c-form__field c-form__field--password"
            >
              <header
                class="c-form__field-head"
              >
                <label
                  for="password"
                >
                  Password
                </label>
              </header>
              <input
                id="password"
                name="password"
                placeholder="********"
                type="password"
              />
            </div>
            <div
              class="c-form__field c-form__field--password"
            >
              <header
                class="c-form__field-head"
              >
                <label
                  for="confirm"
                >
                  Confirm password
                </label>
              </header>
              <input
                id="confirm"
                name="confirm"
                placeholder="********"
                type="password"
              />
            </div>
            <div
              class="c-form__field c-form__field--text"
            >
              <header
                class="c-form__field-head"
              >
                <label
                  for="username"
                >
                  Username
                </label>
              </header>
              <input
                id="username"
                name="username"
                placeholder="JohnSmith"
                type="text"
              />
            </div>
          </div>
          <div
            class="c-form__col"
          >
            <div
              class="c-form__field c-form__field--text"
            >
              <header
                class="c-form__field-head"
              >
                <label
                  for="first_name"
                >
                  First name
                </label>
              </header>
              <input
                id="first_name"
                name="first_name"
                placeholder="John"
                type="text"
              />
            </div>
            <div
              class="c-form__field c-form__field--text"
            >
              <header
                class="c-form__field-head"
              >
                <label
                  for="last_name"
                >
                  Last name
                </label>
              </header>
              <input
                id="last_name"
                name="last_name"
                placeholder="Smith"
                type="text"
              />
            </div>
            <div
              class="c-form__field c-form__field--select"
            >
              <header
                class="c-form__field-head"
              >
                <label
                  for="gender"
                >
                  Gender
                </label>
              </header>
              <select
                id="gender"
                name="gender"
              >
                <option
                  value="Unspecified"
                >
                  Unspecified
                </option>
                <option
                  value="Male"
                >
                  Male
                </option>
                <option
                  value="Female"
                >
                  Female
                </option>
              </select>
            </div>
            <div
              class="c-form__field c-form__field--url"
            >
              <header
                class="c-form__field-head"
              >
                <label
                  for="image_url"
                >
                  Avatar image URL
                </label>
              </header>
              <input
                id="image_url"
                name="image_url"
                placeholder="https://images.com/john-smith.jpg"
                type="url"
              />
            </div>
          </div>
        </div>
        <button
          class="c-btn "
          type="submit"
        >
          Sign up
        </button>
        <p
          class="c-form__hint"
        >
          Already have an account? 
          <a
            href="/login"
          >
            Login
          </a>
        </p>
      </form>
    </main>
    <footer
      class="c-footer"
    >
      <div
        class="o-shell"
      >
        <p>
          Â© 2023. All rights reserved.
        </p>
        <p>
          Change language:
          <button>
            de
          </button>
          <button
            disabled=""
          >
            en
          </button>
        </p>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
